src/regions.ts

```diff
@@ -70,6 +76,28 @@ function getConnectionURI(region?: Regions | RegionsObj): RegionsObj {
   }
 }
 
+/**
+ * Set region in-memory to be inherited by other modules when set in the Analysis runtime
+ * with `Analysis.use()`.
+ *
+ * @example
+ *
+ * ```ts
+ * async function myAnalysis(context, scope) {
+ *   // this uses the region defined through `use`
+ *   const resources = new Resources({ token });
+ *
+ *   // it's still possible to override if needed
+ *   const europeResources = new Resources({ token, region: "eu-w1" });
+ * }
+ *
+ * Analysis.use(myAnalysis, { region: "us-e1" });
+ * ```
+ */
+function setRuntimeRegion(region: RegionsObj) {
+  runtimeRegion = region;
```

Comment:
Storing in-memory since modules used in the Analysis function do not inherit from `Analysis.use()`